packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpJesApi.ts

```diff
@@ -0,0 +1,249 @@
+                });
+                return results;
+            }
+        }
+        return [result];
+    }
+
+    public async getJob(jobid: string): Promise<zowe.IJob> {
+        const result = this.getIJobResponse();
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            const jobStatus: IJobStatus = await JobUtils.findJobByID(connection, jobid);
+            if (jobStatus) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                return {
+                    ...result,
+                    /* it’s prepared for the potential change in zftp api, renaming jobid to jobId, jobname to jobName. */
+                    jobid: (jobStatus as any).jobId || jobStatus.jobid,
+                    jobname: (jobStatus as any).jobName || jobStatus.jobname,
+                    owner: jobStatus.owner,
+                    class: jobStatus.class,
+                    status: jobStatus.status,
+                };
+            }
+        }
+        return result;
+    }
+
+    public async getSpoolFiles(jobname: string, jobid: string): Promise<zowe.IJobFile[]> {
+        const result = this.getIJobFileResponse();
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            const response: IJobStatus = await JobUtils.findJobByID(connection, jobid);
+            const files: any = response.spoolFiles;
+            if (files) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                return files.map((file: ISpoolFile) => {
+                    return {
+                        /* it’s prepared for the potential change in zftp api, renaming stepname to stepName, procstep to procStep, ddname to ddName. */
+                        jobid: jobid,
+                        jobname: jobname,
+                        "byte-count": file.byteCount,
+                        id: file.id,
+                        stepname: (file as any).stepName || file.stepname,
+                        procstep: (file as any).procStep || file.procstep,
+                        class: file.class,
+                        ddname: (file as any).ddName || file.ddname,
+                    };
+                });
+            }
+        }
+        return [result];
+    }
+    public async downloadSpoolContent(parms: zowe.IDownloadAllSpoolContentParms): Promise<void> {
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            const destination = parms.outDir == null ? "./output/" : parms.outDir;
+            const jobDetails = await JobUtils.findJobByID(connection, parms.jobid);
+            if (jobDetails.spoolFiles == null || jobDetails.spoolFiles.length === 0) {
+                throw new Error("No spool files were available.");
+            }
+            const fullSpoolFiles = await JobUtils.getSpoolFiles(connection, jobDetails.jobid);
```

Comment:
As we discussed, it's duplicate code with zftp. We may add new job API in the next zftp to cover spool file downloading.

Reply:
Yes, I add a comment in this part. Thank you!